docs/advanced/misc.rst

```diff
@@ -304,3 +304,31 @@ the default settings are restored to prevent unwanted side effects.
 
 .. [#f4] http://www.sphinx-doc.org
 .. [#f5] http://github.com/pybind/python_example
+
+.. _avoid-cpp-types-in-docstrings:
+
+Avoid C++ types in docstrings
+=============================
+
+Most of docstrings are generated at the time of the declaration, e.g. when ``.def(...)`` is called.
+At this point arguments/return types should be known to pybind11, otherwise C++ types would be used instead:
```

Comment:
Maybe "argument and return types"?

What about: "... be known to pybind11, otherwise C++ types would be used > "...  be known to pybind11. If a custom type is not exposed yet through a `py::class_` constructor or a custom type caster, its C++ type name will be used instead to generate the signature in the docstring:"

Reply:
Sounds good to me!